logagg/formatters.py

```diff
@@ -16,6 +16,59 @@ def docker_file_log_driver(line):
                         )
     return dict(timestamp=log.get('timestamp'), data=log, type='log')
 
+def haproxy(line):
```

Comment:
Also seems like you are handling only one type of log. You can try a different approach to store the logs in json format and the write a formatter fo the logs. It will be easy to parse and also you can handle multiple logs if it is structure. Just find out how to configure the log format of Haproxy if possible. We do this for nginx. Ref: https://github.com/deep-compute/logagg/issues/61

Reply:
@supriyopaul As of now, We don't have access to Haproxy log configuration. So for now, let's process like this.